0_2_pandas_dataframes_attributes_methods.Rmd

```diff
@@ -820,6 +1088,60 @@ df['Population_z'].sort_values()
 ... but for a larger dataset we could find them using [Boolean filtering - a topic we will come to in a later page](0_5_filtering_data_with_pandas.Rmd).
 
 
+::: {exercise-start}
+:label: getting-barred
+:class: dropdown
+:::
+
+If you use `.plot(kind = "bar")` you will generate a bar plot (that is, a chart with bars representing counts for different categories).
+
+Your task is to use indexing methods of your choice, along with **only** the Pandas methods shown on this page, to create a bar plot which has:
+
+* the name of each country on the x-axis
+
+...and...
+
+* bars which show the population of the country on the y-axis. The bars should be ordered from highest population (left hand side of the graph) to lowest population (right hand side of the graph).
+
+*Unless you enjoy cheating, do not use other plotting libraries!*. Your final result should look like this:
+
+![](images/bar_example.png)
+
+Try to create the plot using as few lines of code as possible.
+
+```{python}
+# your code below
+```
+
+::: {exercise-end}
+:::
+
+::: {solution-start} getting-barred
+:class: dropdown
+:::
+
+It is possible to create this plot in one line of code, again, by *chaining* Pandas methods together. First, we use direct indexing with the column names to select just those variables. 
+
+Then we use `sort_values()` using the `by = 'Population'` and `ascending = False` arguments to ensure that the population counts are sorted in descending order.
+
+We then chain `.plot(kind='bar')` on the end, and we get the desired plot:
+
+```{python}
+# solution
+df[['Country Name', 'Population']].sort_values(by='Population', ascending=False).plot(kind='bar')
+```
+
+*Food for thought*: what will happen if you reverse the order of `Country Name` and `Population` in the direct indexing operation? Perhaps the order is telling `.plot()` which variable to put on the x-axis and which on the y-axis?
```

Comment:
```suggestion
*Food for thought*: what will happen if you reverse the order of `Country Name` and `Population` in the direct indexing operation? Do you think this will change the x-axis and and y-axis variables on the `.plot()`?
```